### PR TITLE
feat: Integrate jupiter

### PR DESCRIPTION
**Pull request by AndonMitev**

### CODE REVIEW

File: packages/solana/lib/index.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 export { SolanaChainProvider } from './chain/SolanaChainProvider';
 export { SolanaNetworks } from './networks';
+export * as TerraTypes from './types';
```
Comment (kraikov) [inline packages/solana/lib/index.ts]: Solana

**Review by kraikov — CHANGES_REQUESTED** [review]

File: packages/solana/lib/wallet/SolanaWalletProvider.ts [hunk 2]
```diff
@@ -14,10 +14,17 @@ import {
...
+    createAccount,
+    createTransferInstruction,
+    getAccount,
+    getAssociatedTokenAddress,
+    TokenAccountNotFoundError,
+} from '@solana/spl-token';
 import { Connection, Keypair, PublicKey, SystemProgram, Transaction as SolTransaction, TransactionInstruction } from '@solana/web3.js';
 import { mnemonicToSeedSync } from 'bip39';
 import { derivePath } from 'ed25519-hd-key';
+import { SolanaTxRequest } from 'lib/types';
```
Comment (kraikov) [inline packages/solana/lib/wallet/SolanaWalletProvider.ts]: this will not compile

**Review by kraikov — CHANGES_REQUESTED** [review]

File: packages/solana/lib/wallet/SolanaWalletProvider.ts [hunk 3]
```diff
@@ -83,38 +90,54 @@ export class SolanaWalletProvider extends Wallet<Connection, Promise<Keypair>> {
...
+                const [fromTokenAccount, toTokenAccount] = await Promise.all([
+                    getAssociatedTokenAddress(contractAddress, this._signer.publicKey),
+                    getAssociatedTokenAddress(contractAddress, to),
+                ]);
+
+                try {
+                    await getAccount(this.chainProvider.getProvider(), toTokenAccount);
+                } catch (err) {
+                    if (err instanceof TokenAccountNotFoundError) {
+                        await createAccount(this.chainProvider.getProvider(), this._signer, contractAddress, to);
```
Comment (monokh) [inline packages/solana/lib/wallet/SolanaWalletProvider.ts:119]: Does this send an extra transaction from the user account in certain situations? 
If this is the case, it is happening without the user realising? And that feels wrong

Comment (AndonMitev) [reply packages/solana/lib/wallet/SolanaWalletProvider.ts:119]: I agree but thats how the solana in general works, even phantom wallet behave at the same way, they are creating token accounts for you under the hood

Comment (AndonMitev) [reply packages/solana/lib/wallet/SolanaWalletProvider.ts:119]: Also transactions have fixed fee which is `00000000.5 SOL` which is very low low amount in cent value, thats why i assume they are doing this under the hood

File: packages/solana/lib/wallet/SolanaWalletProvider.ts (hunk 2 above)
Comment (AndonMitev) [reply packages/solana/lib/wallet/SolanaWalletProvider.ts]: resolved

**Comment by AndonMitev** [comment]
@kraikov @monokh i have added NFT provider as well

**Pull request merged**